File: Python/urgency_stack.py
def generate_stack(clients):
    # only urgent extractions
    urgencies = [c for c in clients if c.service_type.lower() == "extraction"
                 and c.priority.lower() == "urgent"]
    # sort by date, nearest first
    urgencies.sort(key=lambda c: c.appointment_date)
    return urgencies[::-1]  # stack = last in, first out

def show_stack(stack):
    print("\n--- Urgency Stack ---")
    while stack:
        print("Attending:", stack.pop())

File: Java/UrgencyStack.java
import java.util.*;

public class UrgencyStack {
    private Stack<Client> stack = new Stack<>();

    public void generateStack(List<Client> clients) {
        clients.stream()
                .filter(c -> c.getServiceType().equalsIgnoreCase("Extraction")
                        && c.getPriority().equalsIgnoreCase("Urgent"))
                .sorted(Comparator.comparing(Client::getAppointmentDate))
                .forEach(stack::push);
    }

    public void showStack() {
        System.out.println("\n--- Urgency Stack ---");
        while (!stack.isEmpty()) {
            System.out.println("Attending: " + stack.pop());
        }
    }
}
